Swirl/include/cli/cli.h

```diff
@@ -16,31 +18,27 @@ namespace cli {
         std::string value;
     };	
 
-	
-	constexpr std::string_view HELP_MESSAGE = R"(The Swirl compiler
-
-Usage: Swirl <file-path> [-o] <output> [--debug]
+	const std::string USAGE = R"(The Swirl compiler
+Usage: Swirl <input-file> [flags]
 
 Flags:
-	-h, --help      Show this help message
-	-d, --debug     log out steps of the compilation
-	-o, --output    Output file name
-	-r, --run       Run the compiled file
-	-c, --compiler  C++ compiler to use
 )";
 
+    std::string generate_help(const std::vector<Argument> &flags);
+
 	std::vector<Argument> parse(int argc, const char** argv, const std::vector<Argument>& flags);
+
     /**
-     * This function returns the value of the flag requested from the provided args vector.
-     * If the flag is not supplied, it returns false.
-     * If the flag is supplied, it returns the value of the flag.
-     * If the flag is not required to have a value, it returns true.
-     *
-     * @param flag Requested flag
-     * @param args Vector of arguments
-     * @return The value of the flag
-     */
-    std::variant<std::string, bool>
+    * This function returns the value of the flag requested from the provided args vector.
+    * If the flag is not supplied, it returns false.
+    * If the flag is supplied, it returns the value of the flag.
+    * If the flag is not required to have a value, it returns true.
+    *
+    * @param flag Requested flag
+    * @param args Vector of arguments
+    * @return The value of the flag
+    */
+    std::variant</* The flag value */ std::string, /* The flag is supplied or not */ bool>
```

Comment:
```suggestion
    std::variant<std::string, bool>
```
I don't think if those comments are needed. There is already a docstring.